### PR TITLE
typing(replays): type replays post_process.py

### PR DESCRIPTION
**Pull request by JoshFerge**
added to mypy.ini and tweaked so mypy is happy

### CODE REVIEW

File: src/sentry/replays/post_process.py [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
     """Process the response further into the expected output."""
     return list(generate_restricted_fieldset(fields, generate_normalized_output(response)))
 
 
 def generate_restricted_fieldset(
-    fields: Optional[List[str]],
-    response: List[Dict[str, Any]],
-) -> Generator[None, None, Dict[str, Any]]:
+    fields: list[str] | None,
+    response: Iterator[dict[str, Any]],
```
Comment (asottile-sentry) [inline src/sentry/replays/post_process.py:14]: ~usually `Iterable` is more accurate than `Iterator`

**Review by asottile-sentry — APPROVED** [review]
fixnship

File: src/sentry/replays/post_process.py [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+) -> Iterator[dict[str, Any]]:
```
[the remark this answers is not in the document]
Comment (JoshFerge) [reply src/sentry/replays/post_process.py:15]: https://docs.python.org/3/library/typing.html
was going off of: 
>
Alternatively, annotate your generator as having a return type of either Iterable[YieldType] or Iterator[YieldType]:
````def infinite_stream(start: int) -> Iterator[int]:
    while True:
        yield start
        start += 1
````
which seems also correct. lmk if one is preferred over the other.

[the remark this answers is not in the document]
Comment (asottile-sentry) [reply src/sentry/replays/post_process.py:15]: return types should be the most specific type -- parameters should be the most general type.  your generator function is a Generator

[the remark this answers is not in the document]
Comment (asottile-sentry) [reply src/sentry/replays/post_process.py:15]: yeah that's a special case -- where you have global mutables and want to protect their values.  that does not apply here

**Pull request merged**